rust/serving/src/app/store/cbstore/jetstreamstore.rs

```diff
@@ -171,42 +172,94 @@ impl super::CallbackStore for JetStreamCallbackStore {
+                attempts += 1;
+                while let Some(watch_event) = watcher.next().await {
+                    event_count += 1;
+                    let entry = match watch_event {
+                        Ok(event) => event,
+                        Err(e) => {
+                            tracing::error!(?e, "Received error from Jetstream KV watcher");
+                            continue;
+                        }
+                    };
+
+                    // init key is used to signal the start of processing, the value will be the timestamp.
+                    if entry.key == callbacks_init_key {
+                        tracing::info!(
+                            callbacks_init_key,
+                            "Received event for the init key. Ignoring it"
+                        );
                         continue;
                     }
-                };
 
-                // init key is used to signal the start of processing, the value will be the timestamp.
-                if entry.key == callbacks_init_key {
-                    continue;
-                }
+                    // done key is used to signal the end of processing, we can break the loop
+                    if entry.key == callbacks_done_key {
+                        tracing::info!(
+                            callbacks_done_key,
+                            "Received event for the callback_done_key. Stopping the watcher task"
+                        );
+                        break;
+                    }
 
-                // done key is used to signal the end of processing, we can break the loop
-                if entry.key == callbacks_done_key {
-                    break;
+                    let cbr: Callback = entry
+                        .value
+                        .try_into()
+                        .inspect_err(|err| tracing::error!(?err, "Failed to deserialize callback"))
+                        .expect("Failed to deserialize callback");
+
+                    tx.send(Arc::new(cbr))
+                        .await
+                        .inspect_err(|err| {
+                            tracing::error!(
+                                ?err,
+                                "Failed to send callback details from watcher task to main task"
+                            )
+                        })
+                        .expect("Failed to send callback");
                 }
-
-                let cbr: Callback = entry
-                    .value
-                    .try_into()
-                    .expect("Failed to deserialize callback");
-
-                tx.send(Arc::new(cbr))
-                    .await
-                    .expect("Failed to send callback");
+                if event_count == 0 && attempts < 5 {
+                    tracing::warn!(
+                        callbacks_key,
+                        "Watcher for Jetstream key didn't return any events. Will recreate the watcher in 20ms"
```

Comment:
Watcher can fail in the middle as well `(event_count>0)`, before getting all the events. Should we recreate the watcher if it returns none before the `done.processing` event?